src/platform/ESP32/nimble/BLEManagerImpl.cpp

```diff
@@ -1040,33 +1040,32 @@ CHIP_ERROR BLEManagerImpl::StartAdvertising(void)
 
     {
         err = ble_gap_adv_start(own_addr_type, NULL, BLE_HS_FOREVER, &adv_params, ble_svr_gap_event, NULL);
-        if (err != CHIP_NO_ERROR)
+        if (err == BLE_HS_EALREADY)
         {
-            ChipLogError(DeviceLayer, "ble_gap_adv_start() failed: %s", ErrorStr(err));
-            ExitNow();
+            /* This error code indicates that the advertising is already active. Stop and restart with the new parameters */
+            ChipLogProgress(DeviceLayer, "Device already advertising, stop active advertisement and restart");
+            err = ble_gap_adv_stop();
+            if (err != 0)
+            {
+                ChipLogError(DeviceLayer, "ble_gap_adv_stop() failed: %s, cannot restart", ErrorStr(err));
+                return CHIP_ERROR_INTERNAL;
+            }
+            else
+            {
+                err = ble_gap_adv_start(own_addr_type, NULL, BLE_HS_FOREVER, &adv_params, ble_svr_gap_event, NULL);
+            }
         }
-    }
 
-exit:
-    if (err == BLE_HS_EALREADY && connectable)
-    {
-        ChipLogProgress(DeviceLayer,
-                        "Connectable advertising failed because device was already advertising , stop active advertisement");
-        err = ble_gap_adv_stop();
-        if (err != CHIP_NO_ERROR)
+        if (err == 0)
         {
-            ChipLogError(DeviceLayer, "ble_gap_adv_stop() failed: %s", ErrorStr(err));
+            return CHIP_NO_ERROR;
         }
         else
         {
-            err = ble_gap_adv_start(own_addr_type, NULL, BLE_HS_FOREVER, &adv_params, ble_svr_gap_event, NULL);
-            if (err != CHIP_NO_ERROR)
-            {
-                ChipLogError(DeviceLayer, "Retried adv start; ble_gap_adv_start() failed: %s", ErrorStr(err));
-            }
+            ChipLogError(DeviceLayer, "ble_gap_adv_start() failed: %s", ErrorStr(err));
+            return CHIP_ERROR_INTERNAL;
```

Comment:
Returning `CHIP_ERROR_INTERNAL` as of now. The errors should ideally be mapped to CHIP error codes. Filed an issue here: https://github.com/project-chip/connectedhomeip/issues/4256